Mathlib/GroupTheory/SpecificGroups/Cyclic.lean

```diff
@@ -680,3 +680,95 @@ lemma not_isCyclic_iff_exponent_eq_prime [Group α] {p : ℕ} (hp : p.Prime)
   · exact False.elim <| h_cyc <| isCyclic_of_orderOf_eq_card g <| by aesop
 
 end Exponent
+
+section ZMod
+
+open Subgroup AddSubgroup
+
+variable {G H : Type*}
+
+/-- The isomorphism between `(Multiplicative G) / ⟨.ofAdd g⟩` and `Multiplicative (G / ⟨g⟩)`. -/
+def multiplicativeQuotientZpowersZmultiplesMulEquiv [AddCommGroup G] (g : G) :
+    Multiplicative G ⧸ zpowers (Multiplicative.ofAdd g) ≃* Multiplicative (G ⧸ zmultiples g) :=
+  MulEquiv.refl _
+
+lemma multiplicativeQuotientZpowersZmultiplesMulEquiv_mk_eq [AddCommGroup G] (g : G)
+    (h : Multiplicative G) :
+    multiplicativeQuotientZpowersZmultiplesMulEquiv g (QuotientGroup.mk h) =
+    .ofAdd (QuotientAddGroup.mk h) := rfl
+
+lemma multiplicativeQuotientZpowersZmultiplesMulEquiv_symm_mk_eq [AddCommGroup G] (g : G)
+    (h : G) :
+    (multiplicativeQuotientZpowersZmultiplesMulEquiv g).symm (QuotientAddGroup.mk h) =
+    QuotientGroup.mk (.ofAdd h) := rfl
+
+/-- The kernel of `zpowersHom G g` is equal to the subgroup generated by `orderOf g`. -/
+theorem zpowersHom_ker_eq [Group G] (g : G) :
+    (zpowersHom G g).ker = zpowers (Multiplicative.ofAdd ↑(orderOf g)) := by
+  ext
+  simp_rw [MonoidHom.mem_ker, mem_zpowers_iff, zpowersHom_apply, ← orderOf_dvd_iff_zpow_eq_one,
+    ← ofAdd_zsmul, zsmul_eq_mul', Equiv.apply_eq_iff_eq_symm_apply, Int.cast_id,
+    Multiplicative.ofAdd_symm_eq, dvd_def, eq_comm]
```

Comment:
Slight defeq abuse, but a much shorter proof is:
```suggestion
    (zpowersHom G g).ker = zpowers (Multiplicative.ofAdd ↑(orderOf g)) :=
  congr_arg AddSubgroup.toSubgroup <| zmultiplesHom_ker_eq (Additive.ofMul g)
```
(after swapping the order again)

I think this one is probably ok, because it if does fail it's easy to fix by adding lemmas about the interaction between `ker` and `toSubgroup` etc.